erc20-token-bridge/contracts/Bridge.sol

```diff
@@ -79,72 +112,72 @@ contract Bridge is IBridge {
             _signature.s
         );
 
-        if (!isWrapped) {
-            Token(_depositData.token).transferFrom(
-                msg.sender,
-                address(this),
-                _depositData.amount
-            );
-        } else {
-            Token(wrappedToken).burnFrom(msg.sender, _depositData.amount);
-        }
+        Token(_depositData.token).burnFrom(msg.sender, _depositData.amount);
 
-        emit Deposit(
+        emit Burned(
             msg.sender,
             _depositData.to,
             _depositData.token,
             _depositData.targetBridge,
-            _depositData.amount,
-            block.number
+            _depositData.amount
         );
     }
 
-    function withdraw(
+    function unlock(
         WithdrawData calldata _withdrawData
     )
         external
+        payable
+        override
+        nonReentrant
+        validateFee
+        validateToken(_withdrawData.token)
+        validateAmount(_withdrawData.amount)
     {
-        address wrappedToken = tokenToWrappedToken[_withdrawData.token];
-        bool isWrapped = wrappedToken != address(0);
-
-        if (!isWrapped) {
-            Token(_withdrawData.token).transfer(
-                msg.sender,
-                _withdrawData.amount
-            );
-        } else {
-            Token(wrappedToken).mint(msg.sender, _withdrawData.amount);
-        }
+        Token(_withdrawData.token).transfer(msg.sender, _withdrawData.amount);
 
-        emit Withdraw(msg.sender, _withdrawData.token, _withdrawData.amount);
+        emit Unlocked(msg.sender, _withdrawData.token, _withdrawData.amount);
     }
 
-    function increaseBalance(
-        address to,
-        address token,
-        uint256 amount
-    ) external isBridge {
-        balance[to][token] += amount;
+    function mint(
+        WithdrawData calldata _withdrawData
+    )
+        external
+        payable
+        override
+        nonReentrant
+        validateFee
```

Comment:
You don't need to collect fees when claiming the tokens on the target chain, since you've already charged the uer on the source chain (`lock` function).